pygui/__init__.py

```diff
@@ -527,3 +533,57 @@ def create_ops(current, future):
             apply_op(ops[-1], wip)
 
     return ops
+
+
+def equivalent_functions(a: Callable, b: Callable):
+    """Returns whether function a is equivalent to function b.
+
+    ``functools.partial`` functions are also supported (but only when both
+    argument a and b are partial).
+    """
+    if not hasattr(a, "__code__") or not hasattr(b, "__code__"):
+        if isinstance(a, partial) and isinstance(b, partial):
+            return (
+                a.args == b.args
+                and a.keywords == b.keywords
+                and equivalent_functions(a.func, b.func)
+            )
+
+        return a == b
+
+    return equivalent_code(a.__code__, b.__code__)
+
+
+def equivalent_code(a, b):
+    """Returns True if a and b are equivalent code.
+
+    In order to determine this, a number of properties are compared that
+    should be equal between similar functions.
+    It checks all co_* props of __code__ (as seen in Python 3.9), except for:
+      * co_varnames
+      * co_firstlineno
+      * co_lnotab
+      * co_name
+    because those vars can differ without having any impact on the equivalency
+    of the functions themselves.
+    """
+    for attr in [
+        "co_argcount",
+        "co_cellvars",
+        "co_code",
+        "co_consts",
+        "co_filename",
+        "co_flags",
+        "co_freevars",
+        "co_kwonlyargcount",
+        "co_names",
+        "co_nlocals",
+        "co_posonlyargcount",
+        "co_stacksize",
+    ]:
+        attr_a = getattr(a, attr, None)
+        attr_b = getattr(b, attr, None)
+        if attr_a != attr_b:
+            return False
+
+    return True
```

Comment:
just checking: is re-evaluating this on every render really more efficient than re-attaching event handlers on every render?

Reply:
Yes, apparently it still outperformed removing the handler and attaching the new one.

Reply:
I tested this on the pygfx example, but the actual answer should be: it depends :) It depends on the specific renderer.